pallets/liquidity-rewards/src/lib.rs

```diff
@@ -206,54 +154,33 @@ pub mod pallet {
 	#[pallet::generate_store(pub(super) trait Store)]
 	pub struct Pallet<T>(_);
 
+	/// Contains the timestamp in blocks when the current epochs finalized.
+	//
 	// Although this value could be stored inside `EpochData`,
 	// we maintain it separately to avoid deserializing the whole EpochData struct each `on_initialize()` call.
 	// EpochData could be relatively big if there many groups.
-	// We dont have to deserialize the whole struct 99% of the time assuming a duration of 100 blocks,
+	// We dont have to deserialize the whole struct 99% of the time (assuming a duration of 100 blocks),
 	// we only need to perform that action when the epoch finalized, 1% of the time.
 	#[pallet::storage]
-	pub(super) type EndOfEpoch<T: Config> = StorageValue<
-		_,
-		EpochTimestamp<T::BlockNumber>,
-		ValueQuery,
-		FirstEpochTimestamp<frame_system::Pallet<T>, T::InitialEpochDuration>,
-	>;
+	pub(super) type EndOfEpoch<T: Config> =
+		StorageValue<_, T::BlockNumber, ValueQuery, T::InitialEpochDuration>;
 
+	/// Data associated to the current epoch.
 	#[pallet::storage]
-	pub(super) type ActiveEpochData<T: Config> = StorageValue<
-		_,
-		EpochData<T::BlockNumber, T::Balance, T::GroupId, T::Weight, T::MaxGroups>,
-		ValueQuery,
-	>;
+	pub(super) type ActiveEpochData<T: Config> = StorageValue<_, EpochData<T>, ValueQuery>;
 
+	/// Pending update data used when the current epoch finalizes.
+	/// Once it's used for the update, it's reset.
 	#[pallet::storage]
-	pub(super) type NextEpochChanges<T: Config> = StorageValue<
-		_,
-		EpochChanges<
-			T::BlockNumber,
-			T::Balance,
-			T::GroupId,
-			T::CurrencyId,
-			T::Weight,
-			T::MaxChangesPerEpoch,
-		>,
-		ValueQuery,
-	>;
+	pub(super) type NextEpochChanges<T: Config> = StorageValue<_, EpochChanges<T>, ValueQuery>;
```

Comment:
Not because `#[scale_info(skip_type_params(T))]` skip that 😃

Reply:
Also, `Config` (or any generic bound) contains only information at compile time. It never increments the size of the objects

Reply:
Im a bit late to the party but wanted to share that even though the current design is leaner, Substrate actually recommends to use the former, e.g., to specifically declare all required generic parameters instead of pulling from the Config. I suppose one argument is that its a bit more opaque. Couldn't quickly find the PR with the comment about that. The skip macro resulted just from the inconvenience that the compiler else throws when pulling from the Config. 

Definitely no need to change this now but good to keep it in mind for the future and other parts of the code we touch. Or do you disagree @lemunozm ?

Reply:
If you finally find that thread I would be happy to read it! Thanks for sharing this.

From my understanding of this, I think that if the type is (or should) be agnostic of the pallet, it shouldn't make use of `<T: Config>`, because it comes with the whole pallet dependency. But sometimes, the type is closely related to the pallet, and it contains a lot of types specified in it. Not using `<T: Config>` in that case adds a lot of boilerplate that seems like it's not really needed and difficult refactors and readability. Anyways, I do not have a strong thought about where to put the frontier between one decision or the other.

P.S: I took as a reference the Substrate's `nomination-pools` pallet where high-level types use `<T: Config>` and low-level types define the generics individually.